src/cmd/play.rs

```diff
@@ -1,30 +1,53 @@
+use super::scan;
 use crate::config;
+use crate::converters;
 use crate::players::shmup;
 use clap::Args;
 
 #[derive(Args, Debug)]
 pub struct RunArgs {
-    /// the path to the directory to use to generate
-    /// the world.
-    /// example:
-    /// `wake play -d ./tmp/github-com-elhmn-ckp`
-    #[arg(short, long, value_name = "DIRECTORY")]
-    dir: Option<String>,
+    /// the path to the repository we want to play
+    #[clap(value_name = "REPOSITORY", index = 1)]
+    repository: Option<String>,
 }
 
-pub fn run(args: &RunArgs, _conf: config::Config) {
-    let dir = args.dir.clone().unwrap_or_default();
-    println!("Play run command invoked");
+pub fn run(args: &RunArgs, conf: config::Config) {
+    let repo = args.repository.clone().unwrap_or_default();
 
-    if !dir.is_empty() {
-        println!("Called with {dir}");
-    }
+    let git_repo = match scan::clone_repository(&repo, &conf) {
+        Ok(r) => {
+            println!("`{repo}` repository cloned successfully");
+            r
+        }
+        Err(err) => {
+            return println!("Error: {err}");
+        }
+    };
+
+    let extracted_data = match scan::extract_data(&conf, &git_repo) {
+        Ok(d) => d,
+        Err(err) => {
+            println!("Error: failed to extract repository data: {err}");
+            return;
+        }
+    };
+
+    let conv = converters::shmup::new();
+    let converted_data = match scan::convert_data(&conf, &git_repo, extracted_data, &conv) {
+        Ok(d) => d,
+        Err(err) => {
+            println!("Error: failed to convert extracted data: {err}");
+            return;
+        }
+    };
```

Comment:
Couldn't `return println!("Error: {err}");` line 23 be print then return as in other errors line 30 and 39 ?
It would be more homogeneous and the return print disturbed me a bit.